client/JSDebugger/mlDebug.ts

```diff
@@ -175,24 +175,23 @@ export class MLDebugSession extends LoggingDebugSession {
                 return bp
             })
 
-            const newBp = new Set()
+            const newBp: Set<string> = new Set()
             args.breakpoints.forEach(breakpoint => {
                 newBp.add(JSON.stringify({
-                    path: path,
                     line: breakpoint.line,
                     column: breakpoint.column,
                     condition: breakpoint.condition
                 })) //construct the set of new breakpoints, better ways?
             })
 
-            const toDelete = new Set([...this._bpCache].filter(x => !newBp.has(x)))
-            const toAdd = new Set([...newBp].filter(x => !this._bpCache.has(x)))
-            this._bpCache = newBp
+            const toDelete = (path in this._bpCache) ? new Set([...this._bpCache[path]].filter(x => !newBp.has(x))) : new Set()
+            const toAdd = (path in this._bpCache) ? new Set([...newBp].filter(x => !this._bpCache[path].has(x))) : newBp
```

Comment:
This is starting to look pretty complicated. Is there a reason not to just clear all the old breakpoints in ML, and then set the new ones?